src/amo/components/CollectionManager/index.js

```diff
@@ -154,7 +155,22 @@ export class CollectionManagerBase extends React.Component<Props, State> {
     event: ElementEvent<HTMLInputElement | HTMLTextAreaElement>
   ) => {
     event.preventDefault();
-    this.setState({ [event.target.name]: event.target.value });
+    const { name = '', value = '' } = event.target;
+    const { creating } = this.props;
+
+    if (creating && name === 'name' && !this.state.customSlug) {
+      this.setState({
+        slug: value.trim().replace(/[^A-Za-z0-9]/g, '-'),
+        [event.target.name]: event.target.value,
+      });
+    } else if (creating && name === 'slug' && value.trim() !== '') {
+      this.setState({
+        customSlug: true,
+        [event.target.name]: event.target.value,
+      });
+    } else {
+      this.setState({ [event.target.name]: event.target.value });
```

Comment:
maybe you want to use `name` and `value` here since you de-structured `event.target` on line 158.

Reply:
I tried this before and got flow error
<img width="718" alt="screen shot 2018-05-08 at 7 14 05 am" src="https://user-images.githubusercontent.com/5318732/39733317-d124cc44-528f-11e8-94c3-832480421af7.png">

But figured out the error, so yes now it looks cleaner.